From 0b3f58f7502b4693ea9b38c9d3973813dd7c412f Mon Sep 17 00:00:00 2001
From: Yaniv Levinsky <ylevinsky@habana.ai>
Date: Sun, 08 Jan 2023 16:27:37 +0200
Subject: [PATCH] [SW-112281] board: support sve zephyr builds on bins

Add support for sve zephyr builds when creating the .bins files.

When building sve_zephyr package, the dtsi files are located on a clone
of the zephyr project inside the br-e output folder.

By default build based on the dtsi files inside embedded/zephyr.

Change-Id: I2ffbbbd5807747dde19b72bd837990d855c06aaf
Signed-off-by: Yaniv Levinsky <ylevinsky@habana.ai>
---

diff --git a/board/habanalabs/common/image-utility.sh b/board/habanalabs/common/image-utility.sh
index 48f5650..9b2cc2d 100644
--- a/board/habanalabs/common/image-utility.sh
+++ b/board/habanalabs/common/image-utility.sh
@@ -160,8 +160,16 @@
 
 function hl_create_zephyr_bins {
 	local bins_dir=$1
+	zephyr_dir="${BR2_EXTERNAL}/../zephyr"
+
+	# check if the build is for sve zephyr by checking if the sve_zephyr folder exists.
+	# change the zephyr dir path accordingly.
+	if [ -d ${BR2_EXTERNAL}/output/${DEV_NAME}_arc/build/sve_zephyr-custom/zephyr ]; then
+		zephyr_dir="${BR2_EXTERNAL}/output/${DEV_NAME}_arc/build/sve_zephyr-custom/zephyr"
+	fi
+
 	# create config (bins file) for all binaries target
-	for dtsi_file in ${BR2_EXTERNAL}/../zephyr/dts/arc/*; do
+	for dtsi_file in ${zephyr_dir}/dts/arc/*; do
 		target_name=`echo $dtsi_file |  awk -F hl_${DEV_NAME}_ '{print $2}' | awk -F '.dtsi'  '{print $1}'`
 		if [ ! -z "$target_name" ]; then
 			bins_file=${bins_dir}/${HL_TARGET}_${target_name}.bins
